redisson/src/main/java/org/redisson/connection/ConnectionsHolder.java

```diff
@@ -202,9 +204,12 @@ public CompletableFuture<T> acquireConnection(RedisCommand<?> command) {
         CompletableFuture<T> result = new CompletableFuture<>();
 
         CompletableFuture<Void> f = acquireConnection();
-        f.thenAccept(r -> {
-            connectTo(result, command);
-        });
+        // prevent deep stack, switch thread per 100 requests.
+        if (count.incrementAndGet() % 100 == 0) {
+            f.thenAcceptAsync(r -> connectTo(result, command));
```

Comment:
you need to define `serviceManager.getGroup()` as an executor.

Reply:
serviceManager.getGroup().submit(() -> {
                releaseConnection();
});

now can be replaced with only releaseConnection();